kvbc/src/v4blockchain/detail/blocks.cpp

```diff
@@ -0,0 +1,37 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "v4blockchain/detail/blocks.h"
+
+namespace concord::kvbc::v4blockchain::detail {
+
+void Block::addUpdates(const concord::kvbc::categorization::Updates& category_updates) {
+  ConcordAssert(isValid_);
+  static thread_local std::vector<uint8_t> updates_buffer;
+  buffer_.resize(HEADER_SIZE + updates_buffer.size());
+  concord::kvbc::categorization::serialize(updates_buffer, category_updates.categoryUpdates());
+  std::copy(updates_buffer.begin(), updates_buffer.end(), std::back_inserter(buffer_));
```

Comment:
I missed this comment earlier:
Should we move Line no. 21 above Line no. 23 so the three lines will appear as below:
concord::kvbc::categorization::serialize(updates_buffer, category_updates.categoryUpdates());
buffer_.resize(HEADER_SIZE + updates_buffer.size());
std::copy(updates_buffer.begin(), updates_buffer.end(), std::back_inserter(buffer_));

i.e.: The buffer_resize happens after we serialize.

Reply:
yes! great catch, thanks :)